In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


In [2]:
education_districtwise = pd.read_csv('education_districtwise.csv')

In [4]:
education_districtwise.head(5)

DISTNAME STATNAME  BLOCKS  VILLAGES  CLUSTERS  TOTPOPULAT  OVERALL_LI
0   DISTRICT32   STATE1      13       391       104    875564.0       66.92
1  DISTRICT649   STATE1      18       678       144   1015503.0       66.93
2  DISTRICT229   STATE1       8        94        65   1269751.0       71.21
3  DISTRICT259   STATE1      13       523       104    735753.0       57.98
4  DISTRICT486   STATE1       8       359        64    570060.0       65.00

the VILLAGES column indicates how many villages are in each district, the TOTPOPULAT column indicates the population for each district, and the OVERALL_LI column indicates the literacy rate for each district.

Use describe() to compute descriptive stats:

Now that we have a better understanding of the dataset, let's use Python to compute descriptive stats.

 Data professionals use the describe() function as a convenient way to calculate many key stats all at once. For a numeric column, describe() gives you the following output:

count: Number of non-NA/null observations

mean: The arithmetic average

std: The standard deviation

min: The smallest (minimum) value

25%: The first quartile (25th percentile)

50%: The median (50th percentile)

75%: The third quartile (75th percentile)

max: The largest (maximum) value

In [5]:
#concerned with literacy rate
education_districtwise['OVERALL_LI'].describe()

count    634.000000
mean      73.395189
std       10.098460
min       37.220000
25%       66.437500
50%       73.490000
75%       80.815000
max       98.760000
Name: OVERALL_LI, dtype: float64

# **important**
describe() excludes missing values (NaN) in the dataset from consideration. You may notice that the count, or the number of observations for OVERALL_LI (634), is fewer than the number of rows in the dataset (680). Dealing with missing values is a complex issue outside the scope of this course.

You can also use the describe() function for a column with categorical data, like the **STATNAME** column.

For a categorical column, describe() gives you the following output:

count: Number of non-NA/null observations

unique: Number of unique values

top: The most common value (the mode)

freq: The frequency of the most common value

In [6]:
education_districtwise['STATNAME'].describe()

count         680
unique         36
top       STATE21
freq           75
Name: STATNAME, dtype: object

In [7]:
#range values
range_overall_li = education_districtwise['OVERALL_LI'].max() - education_districtwise['OVERALL_LI'].min()
range_overall_li

61.540000000000006